conjure-java-core/src/integrationInput/java/com/palantir/product/EteBinaryServiceEndpoints.java

```diff
@@ -80,20 +74,20 @@ public void register(EndpointRegistry endpointRegistry) {
 
             @Override
             public void handleRequest(HttpServerExchange exchange) throws IOException {
-                AuthHeader authHeader = Auth.header(exchange);
-                InputStream body = BinarySerializers.deserializeInputStream(exchange);
+                AuthHeader authHeader = context.authHeader(exchange);
```

Comment:
this doesn't really seem like a 'context' function

Reply:
I agree that the `ServiceContext` name doesn't quite fit anymore. We can rename in #261, ideas for a better fit?

Reply:
seems different than a naming problem -- what's the intent of the object? it seems pretty muddy, now.

Reply:
The point is to provide implementations of all non-generated logic. Rather than provide a static utility class in lib, we add methods to this interface, which is provided to generated services. As demonstrated in #261 we delete the majority of the -lib module in favor of moving all logic into the runtime.
We could provide separate interfaces for these functions accessible from the context, however that makes the generated code a bit dirtier due to chained accessors.

Reply:
almost all of [ServiceContext](https://github.com/palantir/conjure-java/blob/58d3abc011a77e83a895a293f42c3b9ae33791fd/conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/ServiceContext.java) is serde logic with exception of authHeader, authCookie and instrument. To me, there seem to be 3 distinct concerns:
 - Serialization
 - Parameter extraction for HttpServerExchange (or, more narrowly: Authorization)
 - "instrumentation" -- which seems superfluous given the signature of this method (it gets access to the implementation + its interface class, which is already available to anyone constructing a service

Given that, I'd propose:
 - remove instrument
 - move authHeader and authCookie to a new interface called AuthorizationExtractor
 - rename ServiceContext to SerDe

Reply:
...possibly more correct to call it `UndertowSerDe`

Reply:
In the future we may need to provide functionality that is not strict ser/de, for example tracking safe/unsafe parameters for request logging #255, or supporting asynchronous request processing by returning a `ListenableFuture`. In these cases we cannot plumb through new types without API breaks, so I'd prefer to add methods to the context. Perhaps `UndertowRuntime`?

Reply:
I think if you go with `UndertowRuntime` it'd be good to break it up into parts (serde, async, etc.) so that it doesn't become a dumping ground for stuff. (the code-gen to make `runtime.serde().<call you care about>` isn't a huge deal, IMO)

Reply:
Works for me, I'll put something together.

Reply:
I agree strongly with @markelliot about splitting and nesting specialized runtime components.

In general I have an issue with naming of runtime and lib. Here is a proposal of splitting and naming of conjure-java:
- conjure-generator
- conjure-lib-api
  - conjure-lib-api-serde
  - ...
- conjure-lib-runtime
  - conjure-lib-runtime-serde
  - ...

conjure-lib-api would be shared between the generator and the runtime.

conjure-lib-api would have to respect semantic versioning. 

For a given proejct, the generator, the runtime and dependencies (for instance WC) must rely on the same major version of the API.
Hence, the dependencies should aim to rely on an conjure-Iib-api as low as possible. 
The runtime can be on a higher version than the generator but not the opposite.
So the dependency constraints would become:
conjure-lib-api version: `dependencies <= generator <= runtime`